extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -419,20 +461,25 @@ export class AmpA4A extends AMP.BaseElement {
     this.lifecycleReporter_.sendPing('preAdThrottle');
     this.timerId_ = incrementLoadingAds(this.win);
     return this.adPromise_.then(rendered => {
-      if (rendered instanceof Error) {
-        // If we got as far as getting a URL, then load the ad, but note the
-        // error.
-        if (this.adUrl_) {
-          this.renderViaCrossDomainIframe_(true);
+      if (rendered instanceof Error || !rendered) {
+        // Haven't rendered yet, so try rendering via one of our
+        // cross-domain iframe solutions.
+        if (this.experimentalNonAmpCreativeRenderMethod_ == 'safeframe' &&
```

Comment:
You're right.  But it makes me nervous to depend here on how logic sets up `this.creativeBody_` in some other method.  If we end up changing it elsewhere, and setting `creativeBody_` no longer depends on the header value, we could end up rendering in SafeFrame in spite of what the header says.  I would prefer to be explicit about dependence here, even if it's strictly a redundant condition.

This is all paranoia based on 'what ifs', of course, so if you feel strongly, I'll change it.  But it feels like a pattern that leads to sneaky bugs.